src/Compilers/Core/Desktop/AnalyzerFileReference.InMemoryAssemblyLoader.cs

```diff
@@ -150,14 +150,20 @@ private static Assembly LoadCore(string fullPath)
             /// </summary>
             private static Assembly CurrentDomain_AssemblyResolve(object sender, ResolveEventArgs args)
             {
-                if (args.RequestingAssembly == null)
-                {
-                    return ResolveForUnknownRequestor(args.Name);
-                }
-                else
+                try
                 {
-                    return ResolveForKnownRequestor(args.Name, args.RequestingAssembly);
+                    if (args.RequestingAssembly == null)
+                    {
+                        return ResolveForUnknownRequestor(args.Name);
+                    }
+                    else
+                    {
+                        return ResolveForKnownRequestor(args.Name, args.RequestingAssembly);
+                    }
                 }
+                catch { }
+
+                return null;
```

Comment:
The exception will be gone and never reported. However, if the exception occurred while we were trying to satisfy a load request that only we can satisfy (e.g. for an analyzer dependency) then other things will start breaking very quickly and you'll know you have a problem.